File: 155.最小栈/solution.cpp
# include "..\leetcode.h"

class MinStack {
public:
    stack<int> val_s;
    stack<int> min_s;
    
    MinStack() {
        min_s.push(INT_MAX);
    }
    
    void push(int val) {
        val_s.push(val);
        min_s.push(min(val, min_s.top()));
    }
    
    void pop() {
        val_s.pop();
        min_s.pop();
    }
    
    int top() {
        return val_s.top();
    }
    
    int getMin() {
        return min_s.top();
    }
};

File: 155.最小栈/solution.py
class MinStack:

    def __init__(self):
        self.valStack = []
        self.minStack = [1e32]

    def push(self, val: int) -> None:
        self.valStack.append(val)
        self.minStack.append(min(val, self.minStack[-1]))

    def pop(self) -> None:
        self.valStack.pop()
        self.minStack.pop()

    def top(self) -> int:
        return self.valStack[-1]

    def getMin(self) -> int:
        return self.minStack[-1]